source/aws-ruby-sdk/s3_helper.rb

```diff
@@ -2,6 +2,10 @@
 
 module IdentityIdpFunctions
   class S3Helper
+    def s3_url?(url)
+      URI.parse(url).host.split('.').include?('s3')
```

Comment:
Not sure how concerned we are for accuracy of this check, but alternatively:

```suggestion
      URI.parse(url).host =~ /\.?s3\.amazonaws\.com$/
```

Could maybe do for a test case or two as well?

Reply:
My thinking was, if any subdomain was exactly equal to `s3` then it was probably an actual S3 URL, or at least a fake that was trying hard enough to be an S3 URL that it would count?

For simplicity's sake I'd like to leave this as-is for now, but happy to re-evaluate if we run into issues with it.

I've added a few test cases, including how this can be "fooled" in 57e971bce435319dd71fbaf720942a1134e3d467